cachecontrol/controller.py

```diff
@@ -90,21 +111,21 @@ def cached_request(self, request):
         Return a cached response if it exists in the cache, otherwise
         return False.
         """
-        cache_url = self.cache_url(request.url)
-        logger.debug('Looking up "%s" in the cache', cache_url)
         cc = self.parse_cache_control(request.headers)
 
         # Bail out if the request insists on fresh data
         if 'no-cache' in cc:
             logger.debug('Request header has "no-cache", cache bypassed')
             return False
 
-        if 'max-age' in cc and cc['max-age'] == 0:
+        if 'max-age' in cc and cc['max-age'] == '0':
```

Comment:
Before fixing this bug I checked how they implement the header parsing and they don't do anything to cast to ints so unless we modify that ourselves only strings will ever come back. That said, max-age is not accessed consistently - it's a combination of this string/int comparison mistake, trying to cast to int and catching the error and an explicit `isdigit` check so I'll make a sweep now and handle it consistently everywhere